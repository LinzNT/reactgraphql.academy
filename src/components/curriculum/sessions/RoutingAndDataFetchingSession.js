import React from 'react'
import Ul, { Li } from '../../layout/Ul'
import Session from './Session'

const RoutingAndDataFetchingSession = ({ title }) => (
  <Session title={title}>
    <Ul>
      <Li>
        React Router v4
        <Ul>
          <Li>Declarative routing</Li>
          <Li>Implementing a master-detail web application</Li>
        </Ul>
      </Li>
      <Li>Component lifecycle</Li>
      <Li>
        Data management
        <Ul>
          <Li>Presentational Components Vs. Container Components</Li>
          <Li>Data fetching</Li>
        </Ul>
      </Li>
    </Ul>
  </Session>
)

export default RoutingAndDataFetchingSession
