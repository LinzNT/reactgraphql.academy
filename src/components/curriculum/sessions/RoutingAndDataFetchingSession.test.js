import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import RoutingAndDataFetchingSession from './RoutingAndDataFetchingSession'

describe('<RoutingAndDataFetchingSession />', () => {
  it('should render component', () => {
    const wrapper = shallow(<RoutingAndDataFetchingSession />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
