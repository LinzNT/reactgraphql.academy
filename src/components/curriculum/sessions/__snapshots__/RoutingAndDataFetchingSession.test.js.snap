// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoutingAndDataFetchingSession /> should render component 1`] = `
<Session>
  <styled.ul>
    <styled.li>
      React Router v4
      <styled.ul>
        <styled.li>
          Declarative routing
        </styled.li>
        <styled.li>
          Implementing a master-detail web application
        </styled.li>
      </styled.ul>
    </styled.li>
    <styled.li>
      Component lifecycle
    </styled.li>
    <styled.li>
      Data management
      <styled.ul>
        <styled.li>
          Presentational Components Vs. Container Components
        </styled.li>
        <styled.li>
          Data fetching
        </styled.li>
      </styled.ul>
    </styled.li>
  </styled.ul>
</Session>
`;
